### DISCUSSION TITLE
Add Optional LDAP authentication

### DISCUSSION THREAD

**Discussion by acherla** [StreetLamb/tribe#125] [Ideas]
Add optional parameters for enabling LDAP authentication for TRIBE using LDAP3
- Ability for user to toggle between default authentication vs LDAP authentication (AUTHENTICATION_METHOD parameter, defaults to the current implementation) - Options (default, ldap) - can add more options as needed
- Ability for user to specify the following configuration options if LDAP is enabled
- LDAP SERVER/PORT
- LDAP search base
- LDAP search filter
- LDAP TLS enablement
- LDAP Cert path
- LDAP TLS path

I can optionally take this PR

**Comment by StreetLamb** [comment]
Hi @acherla, I'm not very familiar with LDAP. What are the benefits of having this as an option?

**Reply by acherla** [reply]
Its normally used by enterprises that have an active directory management system for providing groups or organizations within a company access to a system.
